src/expm.jl

```diff
@@ -0,0 +1,64 @@
+@inline expm(A::StaticMatrix) = _expm(Size(A), A)
+
+@generated function _expm(::Size{(1,1)}, A::StaticMatrix)
+    @assert size(A) == (1,1)
+    T = promote_type(typeof(sqrt(one(eltype(A)))), Float32)
```

Comment:
Moving
```julia  
        T = promote_type(typeof(exp(zero(eltype(A)))), Float32)
        newtype = similar_type(A,T)
```
into the body removes the need for using a generated functions in the first place, right?